### PR TITLE
Dependency verification option for plain-text only

### LINKED ISSUE #20834: Add an option to only generate plaintext and not binary for `--export-keys`
For dependency verification, would like to be only generate `gradle/verification-keyring.keys` and not generate `gradle/verification-keyring.gpg`

related docs: 
https://docs.gradle.org/current/userguide/dependency_verification.html#sec:local-keyring

### Expected Behavior

- `--export-keys text` only generates `gradle/verification-keyring.keys`
- `--export-keys gpg` only generates `gradle/verification-keyring.gpg`
- `--export-keys gpg,text` generates both
- `--export-keys` generates both

### Current Behavior

`--export-keys` always generates both `gradle/verification-keyring.keys` and `gradle/verification-keyring.gpg`

### Context

Our use case is that we want to only use `gradle/verification-keyring.keys` so that we can see changes in the pull request diff. 
We added `gradle/verification-keyring.gpg` to `.gitignore` , but if someone does `--export-keys` on branch A and then changes over to branch B, the `gradle/verification-keyring.gpg` file might cause issues because gradle ignores the `.keys` file if the `.gpg` file is present.

### PR DESCRIPTION
**Pull request by ahmedehabb**
Fixes #20834 

### Context

Many users could benefit from this change because it addresses a specific use case in which developers want to track changes to the gradle/verification-keyring.keys file in pull request differentials. By making it possible to generate only the gradle/verification-keyring.keys file and not the gradle/verification-keyring.gpg file using the --export-keys option and vice versa if people want to keep .gpg only and not .keys file

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [x] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [ ] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### CODE REVIEW

File: subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/artifacts/verification/serializer/DependencyVerificationsXmlReaderTest.groovy [hunk 1]
```diff
@@ -188,6 +188,36 @@ class DependencyVerificationsXmlReaderTest extends Specification {
         verifier.configuration.ignoredKeys == [key("ABCDEF"), key("012345", "nope")] as Set
     }
 
+    def "can parse keyring format"() {
```
Comment (ov7a) [inline subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/artifacts/verification/serializer/DependencyVerificationsXmlReaderTest.groovy:191]: 💅 consider adding a test case with absent keyring-format entry

**Review by ov7a — CHANGES_REQUESTED** [review]
The approach is good overall. Thanks for covering this. 

Please, don't forget to address the errors from sanityCheck.

File: subprojects/docs/src/docs/release/notes.md [hunk 2]
```diff
@@ -104,6 +104,28 @@ versionCatalogs.named("libs").findLibrary("assertj-core").ifPresent { assertjCor
 
 Check the [version catalog API](javadoc/org/gradle/api/artifacts/VersionCatalog.html) for all supported methods.
 
+<a name="dependency-verification"></a>
+
+### Dependency verification improvements
+
+When Gradle is asked to export all trusted keys via `./gradlew --export-keys`, it generates both binary and ASCII armored versions of the keys.
```
Comment (lkasso) [inline subprojects/docs/src/docs/release/notes.md:111]:
```suggestion
When Gradle is asked to export all trusted keys via `./gradlew --export-keys`, it generates both binary and ASCII armored versions of the keys.

```

File: subprojects/docs/src/docs/release/notes.md [hunk 3]
```diff
@@ -104,6 +104,28 @@ versionCatalogs.named("libs").findLibrary("assertj-core").ifPresent { assertjCor
...
+With this release you can change this behavior and choose only one format.
+In order to do so, you can edit `verification-metadata.xml` by adding the `keyring-format` setting.
```
Comment (lkasso) [inline subprojects/docs/src/docs/release/notes.md:113]:
```suggestion
To do so, edit `verification-metadata.xml` by adding the `keyring-format` setting:
```

File: subprojects/docs/src/docs/release/notes.md [hunk 4]
```diff
@@ -104,6 +104,28 @@ versionCatalogs.named("libs").findLibrary("assertj-core").ifPresent { assertjCor
 
 Check the [version catalog API](javadoc/org/gradle/api/artifacts/VersionCatalog.html) for all supported methods.
 
+<a name="dependency-verification"></a>
+
+### Dependency verification improvements
+
+When Gradle is asked to export all trusted keys via `./gradlew --export-keys`, it generates both binary and ASCII armored versions of the keys.
+With this release you can change this behavior and choose only one format.
```
Comment (lkasso) [inline subprojects/docs/src/docs/release/notes.md:112]:
```suggestion
You can now change this behavior to choose only one format.
```

File: subprojects/docs/src/docs/release/notes.md [hunk 5]
```diff
@@ -104,6 +104,28 @@ versionCatalogs.named("libs").findLibrary("assertj-core").ifPresent { assertjCor
...
+                       xsi:schemaLocation="https://schema.gradle.org/dependency-verification https://schema.gradle.org/dependency-verification/dependency-verification-1.3.xsd">
+    <configuration>
+        <verify-metadata>true</verify-metadata>
+        <verify-signatures>true</verify-signatures>
+        <keyring-format>text</keyring-format> <!-- or gpg -->
+    </configuration>
+</verification-metadata>
+```
+
+We've also updated the XML validation schema for `verification-metadata.xml` to be more concise.
```
Comment (lkasso) [inline subprojects/docs/src/docs/release/notes.md:128]: Could be nice to get a little more information or an example of this. How is it more concise?

**Review by lkasso — APPROVED** [review]
Some changes and feedback on release notes.

**Review by ov7a — APPROVED** [review]

**Review by ljacomet — CHANGES_REQUESTED** [review]
Looks good overall to me.

There is one piece missing IMO.

`BuildTreeDefinedKeys` should take into account the new configuration option and only load the relevant file if this is configured.
Otherwise, the current default should remain: `gpg` over `keys`

**Review by ljacomet — APPROVED** [review]

**Comment by ov7a** [comment]
@bot-gradle test and merge

**Pull request merged**